src/mainmenu/plugin.ts

```diff
@@ -70,43 +89,31 @@ namespace MainMenu {
 
 
 /**
- * The main menu extension.
- *
- * #### Notes
- * The main menu extension adds a menu bar to the top area
- * of the application shell.
- *
- */
-export
-const mainMenuExtension = {
-  id: 'jupyter.extensions.main-menu',
-  activate: activateMainMenu
-};
-
-
-/**
  * A service providing an interface to the main menu.
  */
 export
-const mainMenuProvider = {
+const mainMenuProvider: JupyterLabPlugin<MainMenu> = {
   id: 'jupyter.services.main-menu',
-  provides: MainMenu,
-  resolve: () => {
-    return Private.mainMenu;
-  }
+  provides: new Token<MainMenu>('jupyter.services.main-menu'),
+  activate: activateMainMenu
 };
 
 
 /**
  * Activate the main menu extension.
  */
-function activateMainMenu(app: Application): void {
+function activateMainMenu(lab: JupyterLab): MainMenu {
+  Private.menuBar = new MenuBar({ keymap: lab.keymap });
```

Comment:
My point is there should be no need to do this now that we have merged extensions and services, but we can save refactoring for later.